tools/autograd/derivatives.yaml

```diff
@@ -1001,11 +1001,15 @@
   self: soft_margin_loss_backward(grad, self, target, reduction)
 
 - name: relu(Tensor self)
-  self: threshold_backward(grad, self, 0)
+  self: relu_backward(grad, self)
 
 # NB: `output` instead of `self` saves memory. It avoids saving a copy of self.
 - name: relu_(Tensor self)
-  self: threshold_backward(grad, result, 0)
+  self: relu_backward(grad, result)
```

Comment:
I wonder whether it'd have been easier to just implement `threshold_backward` directly for mkldnn and error out if threshold is non zero. That way you'd need to change less of existing code.